contracts/Registries/Overrides.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.11;
+
+import "../Utilities/Ownable.sol";
+
+interface ICurveRegistry {
+    function get_pool_from_lp_token(address) external view returns (address);
+
+    function pool_list(uint256) external view returns (address);
+
+    function pool_count() external view returns (uint256);
+}
+
+interface ICurveAddressesProvider {
+    function max_id() external view returns (uint256);
+
+    function get_address(uint256) external view returns (address);
+}
+
+contract CurveRegistryOverrides is Ownable {
+    ICurveAddressesProvider public curveAddressesProvider;
+    ICurveRegistry public curveRegistry;
+    mapping(address => address) public poolByLpOverride;
+    address[] public poolAddresses;
+
+    constructor(
+        address _curveAddressesProviderAddress,
+        address _curveRegistryAddress
+    ) {
+        require(
+            _curveAddressesProviderAddress != address(0),
+            "Missing Curve Addresses Provider address"
+        );
+        require(
+            _curveRegistryAddress != address(0),
+            "Missing Curve Registry address"
+        );
+        curveAddressesProvider = ICurveAddressesProvider(
+            _curveAddressesProviderAddress
+        );
+        curveRegistry = ICurveRegistry(_curveRegistryAddress);
+    }
+
+    /// @notice Updates Curve Registry Address
+    function addCurveRegistry(address _curveRegistryAddress) public onlyOwner {
+        require(
+            _curveRegistryAddress != address(0),
+            "Missing Curve Registry Address"
+        );
+        curveRegistry = ICurveRegistry(_curveRegistryAddress);
+    }
+
+    /// @notice Returns both override and curve registry pools as list
+    function curveRegistriesList() public view returns (address[] memory) {
```

Comment:
Do we need this function? I don't think a pool is the same thing a registry as well

Reply:
You're right.  Updated to return the curve registries as a list.  Function was an original design requirement, will discuss with @x48115